Source/RapyutaSimulationPlugins/Private/Tools/SimulationState.cpp

```diff
@@ -219,41 +170,27 @@ void ASimulationState::SetEntityStateSrv(UROS2GenericSrv* Service)
     // UE_LOG(LogTemp, Warning, TEXT("SetEntityStateService called - Currently ignoring Twist"));
 
     FROSSetEntityState_Response Response;
-    Response.success = ReferenceFrameToInertiaFrame(Request.state_reference_frame,
-                                                    Request.state_pose_position_x,
-                                                    Request.state_pose_position_y,
-                                                    Request.state_pose_position_z,
-                                                    Request.state_pose_orientation);
-    if (Response.success)
+    Response.success = false;
+
+    if (CheckEntity(Request.state_name, false)) 
     {
-        if (Entities.Contains(Request.state_name))
+        if (CheckEntity(Request.state_reference_frame, true)) 
         {
-            if (IsValid(Entities[Request.state_name]))
-            {
-                AActor* Entity = Entities[Request.state_name];
-                FVector Pos(Request.state_pose_position_x, Request.state_pose_position_y, Request.state_pose_position_z);
-                Entity->SetActorLocationAndRotation(Pos, Request.state_pose_orientation);
-            }
-            else
+            Response.success = true;
+            FVector Pos(Request.state_pose_position_x, Request.state_pose_position_y, Request.state_pose_position_z);
+            FTransform WorldTrans(Request.state_pose_orientation, Pos, FVector::OneVector);

```

Comment:
FTransform's scale already defaults, so this could be written like?
```suggestion
            FTransform WorldTrans(Request.state_pose_orientation,
                                  FVector(Request.state_pose_position_x, Request.state_pose_position_y, Request.state_pose_position_z));
```

Reply:
FTransform's scale already defaults, so this could be written like?
```suggestion
            FTransform WorldTrans(Request.state_pose_orientation,
                                  FVector(Request.state_pose_position_x, Request.state_pose_position_y, Request.state_pose_position_z));
```
& `WorldTransf` with a trailing `f` just to signify itself as a transform, not a translation, similar to various places below.